SonarQube.TeamBuild.PreProcessor/ArgumentProcessor.cs

```diff
@@ -96,11 +96,16 @@ public static ProcessedArgs TryProcessArgs(string[] commandLineArgs, ILogger log
                 Debug.Assert(globalFileProperties != null);
 
                 processed = new ProcessedArgs(
-                    TryGetArgumentValue(KeywordIds.ProjectKey, arguments),
-                    TryGetArgumentValue(KeywordIds.ProjectName, arguments),
-                    TryGetArgumentValue(KeywordIds.ProjectVersion, arguments),
```

Comment:
TryGetArgumentValue was only being called for required arguments i.e. ones that we've already checked exist. GetArgumentValue will now throw if asked for an argument that does not exist.